packages/graph/hash_graph/lib/graph/src/shared/identifier/knowledge.rs

```diff
@@ -134,18 +135,42 @@ impl EntityVersion {
     }
 }
 
-#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, Serialize, Deserialize, ToSchema)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, ToSql)]
+#[postgres(transparent)]
+#[repr(transparent)]
+pub struct EntityRecordId(i64);
+
+impl EntityRecordId {
+    #[must_use]
+    pub const fn new(id: i64) -> Self {
```

Comment:
I wonder if we should go with `from_i64` in this case because any other kind of random ID could be generated in its `new`? No strong feelings about this, but if we want to treat this as a random opaque Id, maybe we should assume it can be generated randomly

Reply:
I expect this mostly to be used by `EntityRecordId::new(row.get(record_id_index))`. I'd agree, but in order to change the generate an ID inside the function this most probably would require removing the `const` anyway. If you feel strongly, I can adjust this in a follow-up PR though.